packages/core/src/browser/dialogs.ts

```diff
@@ -401,6 +401,68 @@ export async function confirmExit(): Promise<boolean> {
     return safeToExit === true;
 }
 
+export class ConfirmDialogSaveProps extends ConfirmDialogProps {
+    readonly save: string;
+    performSave: () => void;
+}
+
+export class ConfirmDialogSave extends ConfirmDialog {
+
+    protected saveButton: HTMLButtonElement | undefined;
+    constructor(
+        @inject(ConfirmDialogSaveProps) protected override readonly props: ConfirmDialogSaveProps
+    ) {
+        super(props);
+        this.contentNode.appendChild(this.createMessageNode(this.props.msg));
+        // reorder buttons
+        this.controlPanel.childNodes.forEach(child => this.controlPanel.removeChild(child));
+        [this.acceptButton, this.closeButton].forEach(child => {
+            if (typeof child !== 'undefined') {
+                this.controlPanel.appendChild(child);
+            }
+        });
+        this.appendSaveButton(props.save).addEventListener('click', () => { props.performSave(); this.acceptButton?.click(); });
```

Comment:
We should break up the callback for clarity:

```suggestion
        this.appendSaveButton(props.save).addEventListener('click', () => { 
        props.performSave(); 
        this.acceptButton?.click(); 
     });
```